src/main/java/io/stargate/sgv2/jsonapi/service/operation/tasks/DBTaskPage.java

```diff
@@ -0,0 +1,93 @@
+package io.stargate.sgv2.jsonapi.service.operation.tasks;
+
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResultBuilder;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.Operation;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * Base for a page of {@link DBTask}s that have been run, and we want to get a page of results from.
+ *
+ * <p>Implements the {@link Supplier} interface to provide a {@link CommandResult}, which is what is
+ * returned by an {@link Operation}. Subclasses should normally override the {@link
+ * #buildCommandResult()} method, calling the base normally to add the errors and warnings to the
+ * result.
+ *
+ * <p>Implements some re-usable logic for building the {@link CommandResult} .
+ */
+public abstract class DBTaskPage<TaskT extends DBTask<SchemaT>, SchemaT extends SchemaObject>
```

Comment:
First thought it's odd that a (result) Page has refs to Tasks. But maybe some other term than "page" would make sense, assuming they are sets of results (or, result builders?).
Not sure what, just thinking that as a term it's not super clear.
("page of what?")

Reply:
good questions, this has been there from the beginning and can be rethought. 

The idea was:The Operation runs the Tasks, when they complete (success or failure) they are sent to the Accumulator, when all  tasks are complete we ask the Accumulator for the result - which is called the Page. 

We probably had that name because to avoid using "Result" when we have CommandResult. And that for things like a read it makes sense, there is a page of results. 

Something to keep iterating on